src/main/java/org/example/marketplacebackend/controller/WatchlistController.java

```diff
@@ -1,8 +1,110 @@
 package org.example.marketplacebackend.controller;
 
-//@RequestMapping("/v1/endpoint-name")
-//@CrossOrigin(origins = {"http://localhost:3000, https://marketplace.johros.dev"}, allowCredentials = "true")
-//@Controller
+import jakarta.transaction.Transactional;
+import java.security.Principal;
+import java.util.List;
+import java.util.UUID;
+import org.example.marketplacebackend.DTO.incoming.ProductCategoryDTO;
+import org.example.marketplacebackend.DTO.outgoing.WatchListResponseDTO;
+import org.example.marketplacebackend.model.Account;
+import org.example.marketplacebackend.model.ProductCategory;
+import org.example.marketplacebackend.model.Watchlist;
+import org.example.marketplacebackend.repository.ProductCategoryRepository;
+import org.example.marketplacebackend.repository.WatchListRepository;
+import org.example.marketplacebackend.service.UserService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequestMapping("/v1/watchlist")
+@CrossOrigin(origins = {
+    "http://localhost:3000, https://marketplace.johros.dev"}, allowCredentials = "true")
+@RestController
 public class WatchlistController {
 
-}
+  private final UserService userService;
+  private final WatchListRepository watchListRepository;
+  private final ProductCategoryRepository productCategoryRepository;
+
+  public WatchlistController(UserService userService, WatchListRepository watchListRepository,
+      ProductCategoryRepository productCategoryRepository) {
+    this.userService = userService;
+    this.watchListRepository = watchListRepository;
+    this.productCategoryRepository = productCategoryRepository;
+  }
+
+  @GetMapping("")
+  public ResponseEntity<?> getWatchList(Principal user) {
+    Account authenticatedUser = userService.getAccountOrException(user.getName());
+
+    List<Watchlist> watchList = watchListRepository.findAllBySubscriber(authenticatedUser);
+
+    List<WatchListResponseDTO> allWatchLists = watchList.stream()
+        .map(list -> new WatchListResponseDTO(list.getId(), list.getProductCategory())).toList();
+
+    return ResponseEntity.status(HttpStatus.OK).body(allWatchLists);
+  }
+
+  @PostMapping("")
+  public ResponseEntity<?> postWatchListItem(Principal user,
+      @RequestBody ProductCategoryDTO productCategoryDTO) {
```

Comment:
I still think that an ID is enough as input, as the name attribute is never used, however, this is fine.

Reply:
@Luxcorel What do you think, should we keep it as DTO or replace with ID?

Reply:
I still think that ID is enough, as we don't use the product category name attribute for anything.